x-pack/test/api_integration/deployment_agnostic/default_configs/stateful.config.base.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import Fs from 'fs';
+import {
+  MOCK_IDP_REALM_NAME,
+  MOCK_IDP_ENTITY_ID,
+  MOCK_IDP_ATTRIBUTE_PRINCIPAL,
+  MOCK_IDP_ATTRIBUTE_ROLES,
+  MOCK_IDP_ATTRIBUTE_EMAIL,
+  MOCK_IDP_ATTRIBUTE_NAME,
+} from '@kbn/mock-idp-utils';
+import {
+  esTestConfig,
+  kbnTestConfig,
+  systemIndicesSuperuser,
+  FtrConfigProviderContext,
+} from '@kbn/test';
+import { CA_CERT_PATH } from '@kbn/dev-utils';
+import { services } from '../services';
+
+interface CreateTestConfigOptions {
+  esServerArgs?: string[];
+  kbnServerArgs?: string[];
+  testFiles: string[];
+  junit: { reportName: string };
+  suiteTags?: { include?: string[]; exclude?: string[] };
+}
+
+export function createStatefulTestConfig(options: CreateTestConfigOptions) {
+  return async ({ readConfigFile }: FtrConfigProviderContext) => {
+    const xPackAPITestsConfig = await readConfigFile(require.resolve('../../config.ts'));
+
+    // TODO: move to kbn-es because currently metadata file has hardcoded entityID and Location
+    const idpPath = require.resolve(
+      '@kbn/security-api-integration-helpers/saml/idp_metadata_mock_idp.xml'
+    );
+
+    const servers = {
+      kibana: {
+        ...kbnTestConfig.getUrlParts(systemIndicesSuperuser),
+        protocol: process.env.TEST_CLOUD ? 'https' : 'http',
+        certificateAuthorities: process.env.TEST_CLOUD
+          ? undefined
+          : [Fs.readFileSync(CA_CERT_PATH)],
+      },
+      elasticsearch: {
+        ...esTestConfig.getUrlParts(),
+        protocol: process.env.TEST_CLOUD ? 'https' : 'http',
+        certificateAuthorities: process.env.TEST_CLOUD
+          ? undefined
+          : [Fs.readFileSync(CA_CERT_PATH)],
+      },
```

Comment:
__question:__ do we really need these `certificateAuthorities` for Kibana and ES in stateful tests? I thought we don't use TLS/HTTPS when running those on CI or locally.

Reply:
ah, my bad. Totally agree.